run.py

```diff
@@ -2,10 +2,13 @@
 import logging
 import os
 
+from checkstorage import checkStorage
+
 def main():
     logging.basicConfig(level=logging.INFO)
     logging.info("Running")
 
+    checkStorage() # exception if no storage
```

Comment:
This seems unnecessary

Reply:
The reason for this is that otherwise storage happens when plugins are loaded, and all exceptions are caught and ignored during that phase (by slackbot package). In order to have the app refuse to start up if storage cannot be initialized, we have to ensure it happens here.  I could import store at the top with other imports, but then the module imported isn't used, so I felt it was more clear to wrap this in a function.  It's a little awkward the way it came out though, but we do want to initialize there before starting the "bot" object.

Reply:
Is importing it sufficient though? What if the key is wrong and the S3 auth doesn't connect. At what point will this be obvious?

Reply:
it will throw an exception... but this conversation convinces me to organize it a little differently.  Not doing initialization on import, but create the objects in a factory which can be called prior to handing execution to the slackbot "Bot" object.